build_projects/dotnet-cli-build/UpdateVersionsRepo.cs

```diff
@@ -11,15 +12,21 @@ public class UpdateVersionsRepo : Task
         [Required]
         public string BranchName { get; set; }
 
+        [Required]
+        public string PackagesDirectory { get; set; }
+
+        [Required]
+        public string GitHubPassword { get; set; }
+
         public override bool Execute()
         {
-            string githubAuthToken = EnvVars.EnsureVariable("GITHUB_PASSWORD");
-            string nupkgFilePath = Dirs.Packages;
-            string branchName = BranchName;
-            string versionsRepoPath = $"build-info/dotnet/cli/{branchName}/Latest";
+            string versionsRepoPath = $"build-info/dotnet/cli/{BranchName}";
 
-            VersionRepoUpdater repoUpdater = new VersionRepoUpdater(githubAuthToken);
-            repoUpdater.UpdatePublishedVersions(nupkgFilePath, versionsRepoPath).Wait();
+            GitHubAuth auth = new GitHubAuth(GitHubPassword);
+            GitHubVersionsRepoUpdater repoUpdater = new GitHubVersionsRepoUpdater(auth);
+            repoUpdater.UpdateBuildInfoAsync(
+                new [] { PackagesDirectory }, 
```

Comment:
I wanted to leave the directory searching to the caller (in particular, msbuild) and stay flexible.